sysutils/mise/Portfile

```diff
@@ -19,15 +19,17 @@ long_description    mise is a front-end to your dev environment. \
 categories          sysutils
 license             MIT
 
-maintainers         {outlook.com:gjq.uoiai @MisLink} openmaintainer
+maintainers         {outlook.com:gjq.uoiai @MisLink} \
+                    {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
 
 checksums           ${distname}${extract.suffix} \
-                    rmd160  376e4eeca4c6538ce8f88f0d18c7e320e8e7e68b \
-                    sha256  757f4e5a8196a58d50a813306cbd50c25d975a4ed0b87f8ec1188308757f5569 \
-                    size    3051593
+                    rmd160  fcb4197c3ecb42bc843ba4dd8c81acc760891769 \
+                    sha256  e86e277e8a136d331667fccb13b1d44fda320271c8956a7bb13c110857990529 \
+                    size    3059332
 
```

Comment:
I added a patch file that just disables the self update check; I was trying to use the `.disable-self-update` file check that was previously built, but I couldn't get it working (the self-update command still did something instead of erroring out like it should).

This also builds the completion files for bash, fish, and zsh.

```suggestion

patchfiles          patch-src_cli_self_update.diff

post-build {
    set release_dir ${worksrcpath}/target/[cargo.rust_platform]/release
    set mise ${release_dir}/${name}

    set bash_completions_dir ${release_dir}/share/bash-completion/completions
    set fish_completions_dir ${release_dir}/share/fish/vendor_completions.d
    set zsh_completions_dir ${release_dir}/share/zsh/site-functions

    xinstall -d ${bash_completions_dir} ${fish_completions_dir} ${zsh_completions_dir}

    system "${mise} completion bash > ${bash_completions_dir}/${name}"
    system "${mise} completion fish > ${fish_completions_dir}/${name}.fish"
    system "${mise} completion zsh > ${zsh_completions_dir}/_${name}"
}
```

The patch file:

```diff
--- src/cli/self_update.rs.old	2024-08-01 23:51:40
+++ src/cli/self_update.rs	2024-08-01 23:57:12
@@ -103,11 +103,6 @@
     }
 
     pub fn is_available() -> bool {
-        !std::fs::canonicalize(&*env::MISE_BIN)
-            .ok()
-            .and_then(|p| p.parent().map(|p| p.to_path_buf()))
-            .and_then(|p| p.parent().map(|p| p.to_path_buf()))
-            .map(|p| p.join("lib").join(".disable-self-update").exists())
-            .unwrap_or_default()
+        false
     }
 }
```